### PR TITLE
UGENE-7824 Impossible to click the right button on the annotated sequence from Annotation Table

### PR DESCRIPTION
**Pull request by rasputinkirill**

### CODE REVIEW

File: src/corelibs/U2View/src/ov_sequence/AnnotationsTreeView.cpp [hunk 1]
```diff
@@ -1789,7 +1789,9 @@ void AnnotationsTreeView::annotationClicked(AVAnnotationItem* item, QMap<AVAnnot
     Qt::KeyboardModifiers km = QApplication::keyboardModifiers();
     const bool controlOfShiftPressed = km.testFlag(Qt::ControlModifier) || km.testFlag(Qt::ShiftModifier);
     if (!controlOfShiftPressed) {
-        sequenceSelection->clear();
+        if (sequenceSelection->getSelectedRegions().toList() != item->annotation->getRegions().toList()) {
```
Comment (yalgaer) [inline src/corelibs/U2View/src/ov_sequence/AnnotationsTreeView.cpp:1792]: А если ты выделишь с дабл-кликом обе аннотации/последовательности и после этого нажмёшь правую кнопку?
Нужно сравнивать полные выделения.

**Review by yalgaer — CHANGES_REQUESTED** [review]
See the comment above.
Modify the test to handle this case too

File: src/corelibs/U2View/src/ov_sequence/AnnotationsTreeView.cpp [hunk 2]
```diff
@@ -1842,6 +1844,8 @@ void AnnotationsTreeView::annotationDoubleClicked(AVAnnotationItem* item, const
 
     QList<U2Region> regionsToSelect = selectedRegions;
     const QVector<U2Region> regions = sequenceSelection->getSelectedRegions();
+    U2Region selectedRegion = selectedRegions.size() == 1 ? selectedRegions.first() : U2Region();
+    bool clearSelection = !(regions.size() == 1 && regions.first() == selectedRegion);
```
Comment (yalgaer) [inline src/corelibs/U2View/src/ov_sequence/AnnotationsTreeView.cpp]: You compute these values but never use them. Is it a bug?

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 3]
```diff
@@ -4260,6 +4260,39 @@ GUI_TEST_CLASS_DEFINITION(test_7806) {
...
+    // Expected: the corresponding sequence is still selected
+    // Current: sequence selection is gone, only annotation selection left
+
+    GTFileDialog::openFile(os, testDir + "_common_data/scenarios/project/", "1.gb");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    GTTreeWidget::doubleClick(os, GTUtilsAnnotationsTreeView::findItem(os, "B_group  (0, 2)"));
+    GTTreeWidget::doubleClick(os, GTUtilsAnnotationsTreeView::findItem(os, "B"));
+    GTTreeWidget::click(os, GTUtilsAnnotationsTreeView::findItem(os, "B"), -1, Qt::RightButton);
+    GTKeyboardDriver::keyPress(Qt::Key_Escape);
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Use key click here, not key press.

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 4]
```diff
@@ -4260,6 +4260,39 @@ GUI_TEST_CLASS_DEFINITION(test_7806) {
...
+    QVector<U2Region> selection = GTUtilsSequenceView::getSelection(os);
+    CHECK_SET_ERR(selection.size() == 1, "Selection size should be 1, but actual size is " + QString::number(selection.size()));
+    CHECK_SET_ERR(selection.first() == U2Region(29, 91),
+                  QString("Selection doesn't match with 'B' annotation it is (%1, %2) instead of (29, 91).")
+                      .arg(QString::number(selection.first().startPos))
+                      .arg(QString::number(selection.first().length))
+                  );
+    GTTreeWidget::doubleClick(os, GTUtilsAnnotationsTreeView::findItem(os, "C_group  (0, 1)"));
+    GTKeyboardDriver::keyPress(Qt::Key_Control);
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp:4290]: Do not use keyPress around the methods that do not guarantee that keyRelease is called.
Do the following: 
1) Find both widget coordinates to click 
2) keyPress 
3)  double click coordinate 1 (with no "os" so it never fails in the middle)
4)  double click coordinate 2
5) keyRelease.

Or you can add a keyboard modifier to the doubleClick() method and hide press/release there.

**Review by yalgaer — CHANGES_REQUESTED** [review]

File: src/corelibs/U2View/src/ov_sequence/AnnotationsTreeView.cpp (hunk 2 above)
Comment (rasputinkirill) [reply src/corelibs/U2View/src/ov_sequence/AnnotationsTreeView.cpp]: Forgot to remove, it was previous attempt to fix.

**Comment by yalgaer** [comment]
Looks good. Please run GUI tests and I will approve

**Review by yalgaer — APPROVED** [review]

**Pull request merged**